secrets/encrypter.go

```diff
@@ -48,31 +48,31 @@ func newFileSecretSource(file string) secretSource {
 	return &fileSecretSource{fileName: file}
 }
 
-type encrypter struct {
+type Encrypter struct {
 	cipherSuites []cipher.AEAD
 	mux          sync.RWMutex
-	sSource      secretSource
-	closer       chan struct{}
+	SecSource    secretSource
```

Comment:
I think the SecSource and the Closer fields should stay encapsulated. There could be a public interface for the SecretSource, and a `func WithSource(SecretSource) *Encrypter {}` constructor could accept custom implementations when necessary (I see this use case in the secretstest). The closer never needs to be public, the constructor can just initialize the channel, so that it stays hidden and always available when the Close() method is called.

Reply:
done